curvefs/src/mds/schedule/operatorStep.h

```diff
@@ -0,0 +1,166 @@
+/*
+ *  Copyright (c) 2021 NetEase Inc.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+/*
+ * @Project: curve
+ * @Date: 2021-11-8 11:01:48
+ * @Author: chenwei
+ */
+
+#ifndef CURVEFS_SRC_MDS_SCHEDULE_OPERATORSTEP_H_
+#define CURVEFS_SRC_MDS_SCHEDULE_OPERATORSTEP_H_
+
+#include <cstdint>
+#include <string>
+#include "curvefs/src/mds/schedule/topoAdapter.h"
+
+namespace curvefs {
+namespace mds {
+namespace schedule {
+enum ApplyStatus {
+    Finished,
+    Failed,
+    Ordered,
+    OnGoing
+};
+
+/**
+ * @brief OperatorStep is used to abstract different operator step.
+ * An Operator is composed of several OperatorStep.
+ */
+class OperatorStep {
```

Comment:
Also we can use template to reuse OperatorStep

Reply:
That need modify curvebs code. Add todo here, do it later after fs branch merge to master branch.
Add todo here.

Reply:
done